documents/coredocument_test.go

```diff
@@ -488,3 +488,21 @@ func TestCoreDocument_GetSignCollaborators(t *testing.T) {
 	assert.Len(t, cs, 1)
 	assert.Contains(t, cs, id1)
 }
+
+func TestCoreDocument_AddAttribute(t *testing.T) {
+	id1 := testingidentity.GenerateRandomDID()
+	cas := CollaboratorsAccess{ReadWriteCollaborators: []identity.DID{id1}}
+	cd, err := NewCoreDocumentWithCollaborators(nil, cas)
+	assert.NoError(t, err)
+	err = cd.AddAttribute("com.basf.deliverynote.chemicalnumber", Str, "100")
+	assert.NoError(t, err)
+	assert.True(t, len(cd.Attributes) == 1)
```

Comment:
assert.Len?

Reply:
could we also test cases with empty readableKey or nil value?

Reply:
As the `TODO` says below these would be updated when converters are ready. Right now its just a sanity check.

Reply:
👍